pkg/cmd/backuprestore/backupserver.go

```diff
@@ -0,0 +1,80 @@
+package backuprestore
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"k8s.io/klog/v2"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+
+type backupServer struct {
+	schedule string
+	timeZone string
+	enabled  bool
+	backupOptions
+}
+
+func NewBackupServer(ctx context.Context) *cobra.Command {
+	backupSrv := &backupServer{
+		backupOptions: backupOptions{errOut: os.Stderr},
+	}
+
+	cmd := &cobra.Command{
+		Use:   "backup-server",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := backupSrv.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := backupSrv.Run(ctx); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+	backupSrv.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupServer) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&b.enabled, "enabled", false, "enable backup server")
+	fs.StringVar(&b.schedule, "schedule", "", "schedule specifies the cron schedule to run the backup")
+	fs.StringVar(&b.timeZone, "timezone", "", "timezone specifies the timezone of the cron schedule to run the backup")
+
+	cobra.MarkFlagRequired(fs, "enabled")
+}
+
+func (b *backupServer) Validate() error {
+	return nil
+}
+
+func (b *backupServer) Run(ctx context.Context) error {
+	// handle teardown
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	shutdownHandler := make(chan os.Signal, 2)
+	signal.Notify(shutdownHandler, shutdownSignals...)
```

Comment:
Could use `signal.NotifyContext` to simplify this function.
https://pkg.go.dev/os/signal#NotifyContext

Reply:
https://pkg.go.dev/os/signal#NotifyContext takes only 

```
signals ...[os](https://pkg.go.dev/os).[Signal](https://pkg.go.dev/os#Signal)
```

I need to capture the signal in a chan iiuc !!

Reply:
Are you going to be using the signals directly, or are you just using them as a notification to shutdown? If the latter, then NotifyContext would work as you would just do the `<-ctx.Done()`; this would stop you if the parent context is cancelled or if your signal handler caught a signal.

Reply:
yes this makes perfect sense 👍🏽  

fixing ...

Reply:
done